### DISCUSSION TITLE
MVPAlab: A Machine Learning decoding toolbox for multidimensional electroencephalography data

### DISCUSSION THREAD

**Discussion by dlopezg** [dlopezg/mvpalab#3] [General]
MVPAlab is a MATLAB-based and very flexible decoding toolbox for multidimensional electroencephalography and magnetoen- cephalography data. The MVPAlab Toolbox implements several machine learning algorithms to compute multivariate pattern analyses, cross-classification, temporal generalization matrices and feature and frequency contribution analyses. This toolbox has been designed to include an easy-to-use and very intuitive graphic user interface and data representation software, which makes MVPAlab a very convenient tool for those users with few or no previous coding experience. However, MVPAlab is not for beginners only, as it implements several high and low-level routines allowing more experienced users to design their own projects in a highly flexible manner.

<hr /><em>This discussion was created from the release <a href='https://github.com/dlopezg/mvpalab/releases/tag/v1.0.0'>MVPAlab: A Machine Learning decoding toolbox for multidimensional electroencephalography data</a>.</em>

**Comment by ash19971220** [comment]
Mr Dlopezg, MY name is Benjamin Francis. I am very interested in your MVPalab and have started to try it now, but I seem to have some problems in the process of learning. When I selected a subject and followed the tutorial to set up the parameters of the decoded analysis and started the analysis, I found that it did not end mVPA analysis quickly and seemed to take a long time. I wonder if this is due to the CPU and RAM of the computer or some improper operation
![2(YOXZF}S%3M7C12_PGM(GO](https://user-images.githubusercontent.com/81811401/131944082-b83afb53-b310-42b0-9bf7-071dcbc2575b.png)
![8XR72NNLHJ(TDQTC$E8I@_P](https://user-images.githubusercontent.com/81811401/131944086-f53e561b-33a8-4ee1-a219-0a9011919e93.png)
![DVTS%YC3%C}DG~)4W8PYUYD](https://user-images.githubusercontent.com/81811401/131944088-5426ca40-dc30-4fe3-955b-3f1b6a0f33f3.png)
![GK$M(M3RR$9I2{9_C3~%AJ](https://user-images.githubusercontent.com/81811401/131944090-56fefcfb-1266-4958-b47b-e915af416736.png)
![O2$HJME54 CW_MZ(AXCTD 4](https://user-images.githubusercontent.com/81811401/131944094-1490ba27-72c9-4e0e-929b-d9abefd219d3.png)

**Comment by dlopezg** [comment]
Hi Benjamin, 
Thank you for your interest. Everything seems fine on your screenshots. Decoding analyses are usually slow and time-consuming tasks. The computing time depends on several parameters such as the total number of trials or the sampling frequency, among others. 

- How long it takes to complete the entire analysis? Please share the output with the timestamps.
- How big is your dataset (trials per subject) and what is the sampling frequency?

You can speed up the analysis by increasing the step size in the general configuration tab. If the step size is set to 2, MVPAlab do not analyzes the entire trial (point by point) but in steps of two timepoints, which reduces the computation time by half (and reduces the temporal resolution too). 

Additionally you are running the permutation test, which is always time consuming because  the previous decoding analysis is repeated 100 times (using permuted labels). You can disable this permutation test for dummy analyses. 

Please let me know if you need more information.
